### PR TITLE
Revision of the navbar using Json data to keep the code readable

### PR DESCRIPTION
**Pull request by StewartMckenzie**

### CODE REVIEW

File: src/components/Navbar/Navbar.js [hunk 1]
```diff
@@ -1,132 +1,129 @@
...
-  const [click, setClick] = useState(false);
-  const handleClick = () => setClick(!click);
-  const closeMobileMenu = () => setClick(false);
+import '../../scss/Navbar.scss';
+import navLinkData from './linksData';
 
-  const hideMobile = () => {
+const Navbar = (props) => {
+  //click based functionality
+  const [clicked, setClicked] = useState(false);
```
Comment (DandyReyes) [inline src/components/Navbar/Navbar.js:9]: What does setClicked do...? May you choose a more descriptive name for this variable please.

Comment (DandyReyes) [reply src/components/Navbar/Navbar.js:9]: I don't mind if the variable name is long too, readability over cleverness :)

File: src/components/Navbar/Navbar.js [hunk 2]
```diff
@@ -1,132 +1,129 @@
...
-
-const useOutsideMobileNav=(mobileNavRef) =>{
+  const useOutsideMobileNav = (mobileNavRef) => {
     useEffect(() => {
-      const handleClickOutside=(event)=> {
-        if (mobileNavRef.current && !mobileNavRef.current.contains(event.target)) {
-          setClick(false);
+      const handleClickOutside = (event) => {
+        if (
+          mobileNavRef.current &&
```
Comment (DandyReyes) [inline src/components/Navbar/Navbar.js:23]: This if statement and the if statement on line 41 do the same functionality. Make this a re-usable function.

File: src/components/Navbar/Navbar.js [hunk 3]
```diff
@@ -1,132 +1,129 @@
...
+              <img className='logo' src='./Images/USULogo.svg' alt='' />
+            </Link>
+          )}
+
+          <div className='nav-menu'>
+            {navLinkData.map((linkData, index) => {
+              return (
+                <>
+                  <div className='nav-item'>
+                    {linkData.firstLevel && (
```
Comment (DandyReyes) [inline src/components/Navbar/Navbar.js:90]: This variable name "firstLevel" and "secondLevel" is a bit confusing Please choose a more descriptive name. Just think about you coming back after 20 years to maintain this code or a super young developer maintaining this code.

File: src/components/Navbar/linksData.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
+export const navLinkData = [
```
Comment (DandyReyes) [inline src/components/Navbar/linksData.js:1]: lmao, I just wanted a JSON file but you made a JS file instead. It works all the same. Good job.

File: src/scss/Navbar.scss [hunk 5]
```diff
@@ -26,6 +25,7 @@
 
 .logo {
   height: 50px;
+  z-index: 2;
```
Comment (DandyReyes) [inline src/scss/Navbar.scss:28]: Why is this on the second z-index? I think the first z-index would be sufficient unless something strange happens at that level.

File: src/scss/Navbar.scss [hunk 6]
```diff
@@ -47,8 +47,7 @@
 }
 
 .nav-menu {
-  display: grid;
-  grid-template-columns: repeat(3, auto);
+  display: flex;
```
Comment (DandyReyes) [inline src/scss/Navbar.scss:50]: I don't think this does anything.

File: src/scss/Navbar.scss [hunk 7]
```diff
@@ -75,6 +74,7 @@
 .fas,
 .fab {
   color: $white;
+  font-size: 35px;
```
Comment (DandyReyes) [inline src/scss/Navbar.scss:77]: Make this font-size into a variable please. I see it re-used a couple time already.

File: src/scss/Navbar.scss [hunk 8]
```diff
@@ -137,20 +178,20 @@
...
     width: 100%;
     border-bottom: solid 2px rgba(119, 136, 153, 0.5);
     margin-left: 10vw;
+    z-index: 3;
   }
 
   .nav-links:hover {
-    background-color: $white;
-    color: $mobileHoverText;
+    color: gold;
```
Comment (DandyReyes) [inline src/scss/Navbar.scss:194]: All color should be in a variable. Unless this is the ONLY place you use it.

**Review by DandyReyes — CHANGES_REQUESTED** [review]

File: src/components/Navbar/Navbar.js [hunk 9]
```diff
@@ -5,25 +5,42 @@ import '../../scss/Navbar.scss';
 import navLinkData from './linksData';
 
 const Navbar = (props) => {
-  //click based functionality
-  const [clicked, setClicked] = useState(false);
-  const handleMobileMenu = () => {
-    setClicked(!clicked);
+  //States
+  const [menuActivated, setMenuActivated] = useState(false);
```
Comment (DandyReyes) [inline src/components/Navbar/Navbar.js:9]: For lines 9 to 11, make it into one using state and setState. It just helps consolidate and keep your code cleaner.
```js
const [state, setState] = useState({menuActivated: false, navItemExpanded: false, expandedItem: 
''});
```

File: src/components/Navbar/Navbar.js [hunk 10]
```diff
@@ -5,25 +5,42 @@ import '../../scss/Navbar.scss';
...
+      setExpandedItem(e.target.name);
+      console.log(e.target.name);
+    }
   };
+
+  //Responsive functionality
   const hideMobileOnResize = () => {
     if (window.innerWidth >= 960) {
-      setClicked(false);
+      setMenuActivated(false);
```
Comment (DandyReyes) [inline src/components/Navbar/Navbar.js:28]: when you switch the top code with [state, setState], you could write code that's more concise with 
```js
setState({menuActivated: false, navItemExpanded: false});
```

File: src/components/Navbar/Navbar.js [hunk 11]
```diff
@@ -5,25 +5,42 @@ import '../../scss/Navbar.scss';
...
+  //Mobile Nav menu closure functionality
   const useOutsideMobileNav = (mobileNavRef) => {
     useEffect(() => {
       const handleClickOutside = (event) => {
         if (
           mobileNavRef.current &&
           !mobileNavRef.current.contains(event.target)
         ) {
-          setClicked(false);
+          setMenuActivated(false);
```
Comment (DandyReyes) [inline src/components/Navbar/Navbar.js:42]: Same as previous comment about setState

File: src/components/Navbar/Navbar.js [hunk 12]
```diff
@@ -70,49 +68,104 @@ const Navbar = (props) => {
...
-          <div className='nav-menu'>
-            {navLinkData.map((linkData, index) => {
-              return (
-                <>
-                  <div className='nav-item'>
-                    {linkData.firstLevel && (
-                      <Link className='nav-links ' to={linkData.link}>
-                        {linkData.name}
-                      </Link>
+          <div
```
Comment (DandyReyes) [inline src/components/Navbar/Navbar.js:83]: navLinkData can be destructured. Remember readability!
```js
const { links, nestedLinks, socialLinks } = navLinkData;
```
Also, please use a more descriptive variable than nestedLinks. Please and thank you.

File: src/components/Navbar/Navbar.js [hunk 13]
```diff
@@ -70,49 +68,104 @@ const Navbar = (props) => {
...
+                        onClick={(e) => expandLink(e)}
+                        className={'nav-item'}>
+                        <Link
+                          id='firstLevel'
+                          name={linkData.name.replace(/\s+/g, '')}
+                          className={linkData.class}
+                          to={linkData.link}>
+                          {linkData.name}
+                        </Link>
+                      </div>
```
Comment (DandyReyes) [inline src/components/Navbar/Navbar.js:105]: linkData can be destructured with name, class, link. Please use a more descriptive variable for firstLevel. It's a bit confusing.

File: src/components/Navbar/Navbar.js [hunk 14]
```diff
@@ -70,49 +68,104 @@ const Navbar = (props) => {
...
-                      <Link to={linkData.link}>{linkData.name}</Link>
+                    {linkData.secondLevel && menuActivated && !navItemExpanded && (
+                      <div onClick={(e) => expandLink(e)} className='nav-item'>
+                        <Link
+                          name={linkData.name.replace(/\s+/g, '')}
+                          className={linkData.class}
+                          to={linkData.link}>
+                          {linkData.name}
+                        </Link>
+                      </div>
```
Comment (DandyReyes) [inline src/components/Navbar/Navbar.js:116]: destructure, destructure, destructure! also, need a more descriptive name than secondLevel please.

**Review by DandyReyes — CHANGES_REQUESTED** [review]

File: src/scss/Navbar.scss (hunk 6 above)
Comment (StewartMckenzie) [reply src/scss/Navbar.scss:49]: Removed

File: src/scss/Navbar.scss (hunk 7 above)
Comment (StewartMckenzie) [reply src/scss/Navbar.scss]: Updated

File: src/scss/Navbar.scss (hunk 8 above)
Comment (StewartMckenzie) [reply src/scss/Navbar.scss:167]: Updated

File: src/components/Navbar/Navbar.js [hunk 15]
```diff
@@ -1,132 +1,129 @@
...
-  const handleClick = () => setClick(!click);
-  const closeMobileMenu = () => setClick(false);
+import '../../scss/Navbar.scss';
+import navLinkData from './linksData';
 
-  const hideMobile = () => {
+const Navbar = (props) => {
+  //click based functionality
+  const [clicked, setClicked] = useState(false);
+  const handleMobileMenu = () => {
```
[the remark this answers is not in the document]
Comment (StewartMckenzie) [reply src/components/Navbar/Navbar.js]: Removed

File: src/components/Navbar/Navbar.js (hunk 12 above)
Comment (StewartMckenzie) [reply src/components/Navbar/Navbar.js:83]: Updated

File: src/components/Navbar/Navbar.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (StewartMckenzie) [reply src/components/Navbar/Navbar.js]: Updated

File: src/components/Navbar/Navbar.js (hunk 9 above)
Comment (StewartMckenzie) [reply src/components/Navbar/Navbar.js:9]: Updated

File: src/components/Navbar/Navbar.js (hunk 10 above)
Comment (StewartMckenzie) [reply src/components/Navbar/Navbar.js:28]: Updated

File: src/components/Navbar/Navbar.js (hunk 11 above)
Comment (StewartMckenzie) [reply src/components/Navbar/Navbar.js:42]: Updated

File: src/components/Navbar/Navbar.js (hunk 12 above)
Comment (StewartMckenzie) [reply src/components/Navbar/Navbar.js:83]: Updated

File: src/components/Navbar/Navbar.js (hunk 13 above)
Comment (StewartMckenzie) [reply src/components/Navbar/Navbar.js:105]: Updated

File: src/components/Navbar/Navbar.js (hunk 14 above)
Comment (StewartMckenzie) [reply src/components/Navbar/Navbar.js:116]: Updated

File: src/components/Navbar/Navbar.js [hunk 16]
```diff
@@ -68,11 +77,11 @@ const Navbar = (props) => {
         <div className='spaceHolder'></div>
 
         <div className='nav-menu-container'>
-          {!navItemExpanded && (
+          {!state.navItemExpanded && (
```
Comment (DandyReyes) [inline src/components/Navbar/Navbar.js:80]: you can destructure this as well.
```js
const {navItemExpanded, menuActivated, expandedItem} = state;
```

Comment (DandyReyes) [reply src/components/Navbar/Navbar.js:80]: anywhere that you see

`state`.someVariableName can be destructured

**Review by DandyReyes — CHANGES_REQUESTED** [review]

File: src/scss/Navbar.scss [hunk 17]
```diff
@@ -77,6 +77,24 @@
   height: 15vh;
   width: -webkit-fill-available;
 }
+.social-icon {
+  font-size: 25px;
```
Comment (DandyReyes) [inline src/scss/Navbar.scss:81]: Please use scss variable for ALL the font sizes. I rather change one variable than 20 places on the scss file. Makes our code maintainable. I've seen 25, 30, 35 and 1.8rem size font.

**Review by DandyReyes — CHANGES_REQUESTED** [review]

**Pull request merged**